### DISCUSSION TITLE
CS2 workshop implementation

### DISCUSSION THREAD

**Discussion by EMRandall13** [ReneRebsdorf/CS2-annotations#31] [General]
As of writing, the CS2 Workshop now has full support for annotations.  I wanted to spread awareness as soon as possible and urge the owner to consider migrating all the current work here, to the Workshop.  I'm personally okay with my contributions being left unaccredited.
![Screenshot 2024-12-18 200119](https://github.com/user-attachments/assets/8f946a7d-3ade-4578-ad2f-69aa3e6e8ffe)

**Comment by ReneRebsdorf** [comment]
Thanks for the heads up, I read the update too in excitement

We will migrate this repo to use the workshop feature and keep the code here for everyone to use and for its tests

I will update you
________________________________
Fra: EMRandall13 ***@***.***>
Sendt: Thursday, December 19, 2024 2:02:30 AM
Til: ReneRebsdorf/CS2-annotations ***@***.***>
Cc: Subscribed ***@***.***>
Emne: [ReneRebsdorf/CS2-annotations] CS2 workshop implementation (Discussion #31)


As of writing, the CS2 Workshop now has full support for annotations. I wanted to spread awareness as soon as possible and urge the owner to consider migrating all the current work here, to the Workshop. I'm personally okay with my contributions being left unaccredited.
Screenshot.2024-12-18.200119.png (view on web)<https://github.com/user-attachments/assets/8f946a7d-3ade-4578-ad2f-69aa3e6e8ffe>

**Comment by ReneRebsdorf** [comment]
I created #32 - there is probably more tasks needed as I dive into this.

@EMRandall13 would you like to contribute on this conversion?

**Comment by EMRandall13** [comment]
@ReneRebsdorf I've created a rough thumbnail for the CS2 workshop.  The background will depend on the annotation being presented.
![DoF Test](https://github.com/user-attachments/assets/7b1fefe1-d790-40b9-a5f3-56d75754daf9)

**Reply by ReneRebsdorf** [reply]
Awesome 😊 thats pretty cool

**Comment by EMRandall13** [comment]
[CS2AnnotationWallpaper.zip](https://github.com/user-attachments/files/18266970/CS2AnnotationWallpaper.zip)
Here are the renders for each supported map.  I decided against flavor text because I believe the workshop title is sufficient enough.

**Reply by ReneRebsdorf** [reply]
Awesome, will include in the steam-workshop branch I am working on!

... Do you reckon something could be done to the eyes? :D

**Reply by EMRandall13** [reply]
Sorry for the long wait, didn't get the message.
I replaced the empty eyeball sockets with soccer balls seen in Dust2, as the eyeballs in S2FM fail to render the eyeball shader seen in live matches.  I don't know if there is a way to make the eyeball shader properly render in S2FM, but I'd be more than happy to be proven wrong.

As for the square thumbnails, I am currently working on that as of writing.  Afterwards, I will include screenshots of what I consider to be the highlights of each map.

**Reply by ReneRebsdorf** [reply]
@EMRandall13 I find your workaround with the soccer balls hilarious, nicely thought.
I was thinking of modifying the pictures and inserting the eyes on top, but given the size of the images, I think your solution is fine.

**Comment by ReneRebsdorf** [comment]
@EMRandall13 I updated all images to 720p to fall within the 1mb size limit for thumbnails, the work is beginning to be ready, waiting for approval from valve on the uploads to this collection: https://steamcommunity.com/sharedfiles/filedetails/?id=3397847952

Ideally the pictures should be a square to avoid black bars as can be seen on the link.

Will proceed with the 720p with current eyes for now :)

**Reply by EMRandall13** [reply]
[thumbnail.zip](https://github.com/user-attachments/files/18308339/thumbnail.zip)
These are the thumbnail pics, didn't think much of 'em.

**Reply by EMRandall13** [reply]
[Previews.zip](https://github.com/user-attachments/files/18308425/Previews.zip)
Here are the previews too.

**Reply by ReneRebsdorf** [reply]
Awesome @EMRandall13 

Great idea on the previews, I have uploaded the images and are awaiting approval from valve mod team (They claim to run a scan).
Thumbnails will be updated in a coming PR.

Both will be added to the repo under assets/

If you want, I can add you as contributor on the annotations based on your contributions, if that is of interest we need to connect on steam, my profile can be found here: https://steamcommunity.com/id/zitrez you are free to add me if you wish.

**Reply by EMRandall13** [reply]
I feel like my credits and contributions would be better mentioned inside the repository itself.  I think a text file inside the assets folder would do nicely.

https://steamcommunity.com/id/EMRandall13

**Reply by ReneRebsdorf** [reply]
Will implement according to all-contributors spec based on your input. :-)

@EMRandall13
